exercises/object-manipulation-2/src/pick.js

```diff
@@ -0,0 +1,14 @@
+/* exported pick */
+function pick(source, keys) {
+  var newObject = {};
+  for (var i = 0; i < keys.length; i++) {
+    if (keys[i] in source) {
+      var newValue = source[keys[i]];
+      newObject[keys[i]] = newValue;
+    }
+    if (newValue === undefined) {
+      delete newObject[keys[i]];
+    }
```

Comment:
You are doing this because `in` will give back `true` if a property exists, even if its value is `undefined`. To get around the need for this conditional, you can change your condition above to check if the value of that property is not `undefined`